### PR TITLE
Get rid of locale-aware query string param sorting

### LINKED ISSUE #3: RangeError: Invalid language tag: C
Hello,

I think I found a bug in the `canonParams` function in `duo_sig.js`.  It seems that `canonParams` properly processes a request object with one property, but fails on an object with more than one property.

        var ks = Object.keys(params);
        ks.sort(function(a, b) {
            return a.localeCompare(b, 'C');
        });

When a request object with multiple properties is passed into `jsonApiCall`, like so:

      duo.jsonApiCall('post', '/auth/v2/enroll', {username: user.id, valid_secs: validForSeconds}, function(res) {
    ...
        });

an error is thrown:

    node_modules/duo_api/lib/duo_sig.js:10
        return a.localeCompare(b, 'C');
                 ^
    RangeError: Invalid language tag: C
    at String.localeCompare (native)

Can you take a look at this?  Thank you.

### PR DESCRIPTION
**Pull request by bspoon**
Fixes #3 and the busted tests.  I'm confident this will work 99% of the time.  I'm not sure why this used locale-based sorting in the first place, but I wonder how often '䚚⡻㗐軳朧倪ࠐ킑È셰' will be used as a query parameter name?

### CODE REVIEW

**Comment by ben-duo** [comment]
tl;dr: The issue was not '䚚⡻㗐軳朧倪ࠐ킑È셰' but this PR may be safe anyway.

Duo's server validates authorization headers by generating the same sorted parameter string as the caller. If the server and caller sort parameters in different orders the signature won't match and the request gets rejected.

In some languages (e.g. C#) the program's locale determines whether "ip_whitelist" sorts before "ip_whitelist_enroll_policy" or vice versa. We override locale here to avoid mysterious API errors on systems configured like that. However, I now seem to recall V8 ignores the locale by default?

If we can assume that's always true in Node then, yes, we can drop this. Otherwise we need to force locale-independent sort some other way.

**Comment by bspoon** [comment]
I think `.sort` would probably do the right thing on its own, but just to be sure, how about we compare the strings ourselves using `charCodeAt`?

```
> duo_api@1.0.0 test /Users/Spoon/Code/duo_api_nodejs
> nodeunit tests


duo_sig
✔ Query Parameter Checks - zero params
✔ Query Parameter Checks - one param
✔ Query Parameter Checks - two params
✔ Query Parameter Checks - list string
✔ Query Parameter Checks - printable ascii characters
✔ Query Parameter Checks - unicode fuzz values
✔ Query Parameter Checks - unicode fuzz keys and values
✔ Query Parameter Checks - sort order with common prefix
✔ Query Parameter Checks - sort order with real-world common prefix
✔ Signature Checks - HMAC-SHA1
```

**Comment by mschwager** [comment]
This looks good to me. Any concerns @ben-duo?

If not, @bspoon can you squash the two commits? And I think we're good to go!
